### ISSUE TITLE
Excessive backslashes

### ISSUE THREAD

**Issue by nnposter** [augustd/burp-suite-software-version-checks#11]
Several patterns seem defective due to excessive backslashes. As an example, pattern
```
http://errors\\.angularjs\\.org/([0-9\\.]+)/
```
is not matching on a page that contains `http://errors.angularjs.org/1.4.5/`. When the doubled backslashes are removed, i.e.
```
http://errors\.angularjs\.org/([0-9\.]+)/
```
then the version does get detected correctly.

As a side note, there is no reason to escape "dot" inside a class because it is not considered to be a meta-character. Also, it might be useful to consider using predefined classes, such as `\d` in lieu of `[0-9]`. Therefore a clean pattern would be:
```
http://errors\.angularjs\.org/([\d.]+)/
```
Or, in a pedantic way to avoid matching on a single "dot", such as `http://errors.angularjs.org/./`, the pattern could be
```
http://errors\.angularjs\.org/(\d+(?:\.\d+)*)/
```

**Comment by adetlefsen-rms** [comment]
This issue might be a result of moving the regexes from Java Strings within the class itself to a separate tab-delimited properties file. The file in question is: 

https://github.com/augustd/burp-suite-software-version-checks/blob/master/src/burp/match-rules.tab

By all means, please feel free to send a pull request with any updates to the regex.

**Comment by nnposter** [comment]
Please take a look at https://gist.github.com/nnposter/4b6697f428a83f3de890. All lines in the file have been touched so it does not make sense to send you a patch.

* Some literal "dots" needed escaping.
* All escaping was removed from literal "dots" inside classes.
* Some double escaping needed to be removed.
* Non-referenced groups were converted to non-capturing groups via `(?:`.
* Use of `[0-9]` was converted to `\d`

None of these changes should change the pattern intent. That said, in some cases I was wondering about accuracy of the original pattern. As an example, pattern
```
Microsoft \.NET Framework Version:([\d.]+)
```
does not have a space after the colon. (Again, I am just wondering about it. I do not know for sure that the space should be there.)

Another example the Oracle patterns have open endings via `.*`. Is it to capture the remainder of the line?

Also, what I have not done is to utilize possibly more accurate version pattern `(\d+(?:\.\d+)*)` instead of `([\d.]+)`.  I do not mind updating the patterns accordingly but I am hesitant to do so without your prior consent.

**Comment by augustd** [comment]
Thanks for bringing this up. I've started work on some unit tests to verify the patterns. Once we have that it should be easy to check whether the patterns work as well as add new verifiable patterns to the list. 

If you have instances of actual version numbers from real requests, please send them my way to help develop the tests.

**Comment by augustd** [comment]
Added unit tests and updated the match rules with your changes. All tests pass. That should close this issue.
